source/code/plugins/changetracking_lib.rb

```diff
@@ -136,6 +142,67 @@ def self.transform(inventoryXMLstr, log = nil)
         return ret
     end
 
+    def self.computechecksum(inventory_hash)
+        inventory = {}
+        inventoryChecksum = {}
+
+        if inventory_hash.has_key?("packages")
+           inventory = inventory_hash["packages"]
+        end
+
+        if inventory_hash.has_key?("services")
+           inventory = inventory_hash["services"]
+        end
+
+        if inventory_hash.has_key?("fileInventories")
```

Comment:
Seems like these checks could be better expressed as an if/elsif/else block, and reordering to reflect the ones you want used as default.